libpromises/cf3parse_logic.h

```diff
@@ -1028,4 +1030,58 @@ static inline void ParserBeginBundleBody()
     P.useargs = NULL;
 }
 
+static inline void ParserHandleQuotedListItems()
+{
```

Comment:
I think it's a good idea to set up some of your assumptions here, as assertions. Especially the ones that are quite important for the reader to understand:

```C
assert(P.lval != NULL);
assert(StringEqual(P.lval, "slist") || StringEqual(P.lval, "ilist") || StringEqual(P.lval, "rlist"));
```

Reply:
These exact assertions fail because the parser currently will call the litem rule for more than just ilist/rlist/slist but also inputs like in the test. So these assertions at the top dont work _unless_ i expect this handle to be called only for them. Implying an if check at the parser.
```
litem: QUOTED_STRING
          {
                 if (StringEqual(P.lval, "ilist") || StringEqual(P.lval, "rlist"))
                 {
                         ParserHandleIRlistQuotedStrings();
                 }
                 ...
          }
```
the logic is the same `if i/rlist expand otherwise just pass the string` its just how it is structured.
As i see it the two ways are handle it completely in the Handle or use the Handle only for the irlists.

I will transform the current handle to be more focused on ir/lists and see if it is cleaner.